File: src/main/java/ar/edu/unq/epers/unidad4/model/Gremio.java
package ar.edu.unq.epers.unidad4.model;

import lombok.*;
import org.springframework.data.neo4j.core.schema.GeneratedValue;
import org.springframework.data.neo4j.core.schema.Id;
import org.springframework.data.neo4j.core.schema.Node;

@ToString
@Setter
@Getter
@EqualsAndHashCode
@NoArgsConstructor(access = AccessLevel.PROTECTED)

@Node
public class Gremio {

    @Id
    @GeneratedValue
    private Long id;
    private String nombre;
    private int capacidad;
    private int integrantes = 0;

    public Gremio(String nombre, int capacidad) {
        this.nombre = nombre;
        this.capacidad = capacidad;
    }

    public void retirarIntegrante() {
        if (integrantes > 0)
            integrantes -= 1;
    }

}

File: src/main/kotlin/ar/edu/unq/epers/unidad4/model/Gremio.kt
package ar.edu.unq.epers.unidad4.model

import org.springframework.data.neo4j.core.schema.GeneratedValue
import org.springframework.data.neo4j.core.schema.Id
import org.springframework.data.neo4j.core.schema.Node

@Node
class Gremio() {

    @Id
    @GeneratedValue
    var id: Long? = null
    var nombre: String = ""
    var capacidad: Int = 0
    var integrantes: Int = 0

    constructor(nombre: String, capacidad: Int) : this() {
        this.nombre = nombre
        this.capacidad = capacidad
    }

    fun retirarIntegrante() {
        if (integrantes > 0)
            integrantes -= 1
    }

}